packages/xo-server/src/api/vm.mjs

```diff
@@ -1327,6 +1329,84 @@ importFromEsxi.params = {
   vm: { type: 'string' },
 }
 
+export async function importMultipleFromEsxi({
+  concurrency = 2,
+  host,
+  network,
+  password,
+  sr,
+  sslVerify = true,
+  stopSource = false,
+  stopOnError = false,
+  thin = false,
+  user,
+  vms,
+}) {
+  const task = await this.tasks.create({ name: `importing vms ${vms.join(',')}` })
+  Task.set('total', vms.length)
+  Task.set('done', 0)
+  Task.set('progress', 0)
+  let done = 0
+  return task.run(async () => {
+    const result = {}
+    try {
+      await asyncEach(
+        vms,
+        async vm => {
+          await new Task({ name: `importing vm ${vm}` }).run(async () => {
+            const vmUuid = await this.migrationfromEsxi({
+              host,
+              user,
+              password,
+              sslVerify,
+              thin,
+              vm,
+              sr,
+              network,
+              stopSource,
+            })
+            result[vm] = vmUuid
+          })
+          done++
+          Task.set('done', done)
+          Task.set('progress', Math.round((done * 100) / vms.length))
+        },
+        {
+          concurrency,
+          stopOnError,
+        }
+      )
+      return result
+      // error is an AggregatedError containing all the errors if stopOnError is true, the first error otherwise
+    } catch (error) {
+      // add the successfull VM if the UX want to show them something positive
+      error.succeeded = result
+      throw error
+    }
+  })
+}
+
+importMultipleFromEsxi.params = {
+  concurrency: { type: 'number', optional: true },
+  host: { type: 'string' },
+  network: { type: 'string' },
+  password: { type: 'string' },
+  sr: { type: 'string' },
+  sslVerify: { type: 'boolean', optional: true },
+  stopSource: { type: 'boolean', optional: true },
+  stopOnError: { type: 'boolean', optional: true },
+  thin: { type: 'boolean', optional: true },
+  user: { type: 'string' },
+  vms: {
+    items: {
+      type: 'number',
```

Comment:
there is a strange behaviour of the JSON arguments parsing that cast the id in number (since they are really number) , that are then refused by the parameter checking

Reply:
@fbeauchamp I don't understand the problem you are talking about, are you sure that's not a client's (`xo-web`) issue?

Reply:
in file `packages/xo-cli/index.mjs` line 169 
```js

    if (value.startsWith('json:')) {
      value = JSON.parse(value.slice(5))
      console.log({value})
    }
    ```
    and then launch 
   ` ./index.mjs  vm.importMultipleFromEsxi host=10.10.0.62 user=REDACTED password=REDACTED vms=json:["2","9"] network=6c4e1cdc-9fe0-0603-e53d-4790d1fce8dd  sr=86a9757d-9c05-9fe0-e79a-8243cb1f37f3 sslVerify=false`
   
   then the result is `{ value: [ 2, 9 ] }` not `{ value: [ "2", "9" ] }`
   Also it is confirmed by ajv
   ```
✖ invalid parameters
  property @./vms/0: must be string
  property @./vms/1: must be string
  ```

Reply:
There is an issue in your command line: `json:'["2", "9"]'`, in your case, the quotes were *swallowed* by the shell.

Reply:
thanks @julien-f 
This is the correct ` parameter  vms=json:[\"2\",\"9\"]`

Reply:
Either escape or quote the JSON value using single quotes as in my comment above :slightly_smiling_face: